collaborative-filter.Rmd

```diff
@@ -38,6 +38,8 @@ require("lsa")
 
 #Look at your data, missing values are coded with zero not NA. Investigate cosine similarity and decide whether your missing values should be coded with zeros or NAs. Explain your choice.
 
+#Given that the mathemical function of cosine similary requires vectors to plot, we can't plot values that are NA, and therefore can't take a cosine similarity to them. So we need to leave the zeos in the data set. If we wanted to be careful of how we deal with the zeros we could instead take a centered cosine. 
```

Comment:
I'm not sure what you mean by the term "plot". You need vectors of the same length, that is the issue with using NAs.